public/gameInfo.php

```diff
@@ -89,6 +91,29 @@
 ?>
 <?php if ($gate): ?>
     <?php RenderContentStart($pageTitle) ?>
+    <script>
+        /**
+         * @param {number} currentPreferenceValue
+         * @param {number} matureContentEnumValue
+         * @param {number} gameId
+         */
+        function disableMatureContentWarningPreference(currentPreferenceValue, matureContentEnumValue, gameId) {
+            // eslint-disable-next-line no-bitwise -- this must be calculated via bitwise operations
+            const newPreferencesValue = currentPreferenceValue + (1 << matureContentEnumValue);
```

Comment:
Could you do the bitwise operation at script time (instead of even passing it as parameters)?
```
const newPreferencesValue = <?= $userDetails['websitePrefs'] | (1 << $matureContentPref) ?>;
...
```

Could also inline the `gameId` value as well.

Reply:
Good idea, I will give this a try.

Reply:
This worked and is much cleaner, thanks.